debian/ubuntu-advantage-tools.postinst

```diff
@@ -326,6 +326,16 @@ cleanup_old_motd_files() {
     rm -f $UA_MESSAGES_DIR/motd*
 }
 
+migrate_user_config_post() {
+    # LP: #2004280
+    if [ -f /etc/ubuntu-advantage/uaclient.conf.preinst-backup ]; then
+        # This script modifies the preinst-backup version of the file in-place
+        /usr/bin/python3 /usr/lib/ubuntu-advantage/migrate_user_config.py
```

Comment:
I know this is a corner case scenario, but is it possible for this script to fail and leave `uaclient.conf.preinst-backup` in a poor state ? Because if yes, then we could be creating an invalid config for the user.

Maybe we should have an extra file for the migration and only performs the move command if the script was able to run without issues

Reply:
We have to always `mv` it back because even in failure scenarios it should contain the backed up config.

The code already mitigates the risk that the file is corrupted by doing everything possible, including yaml serialization, before `open`ing and `write`ing the file. So the only possibility of a corrupted file is if the process is killed in  the middle of the `write`. That is possible though.

(Maybe this is what you're suggesting): We could go one step farther and write the new contents to a separate file and `os.rename` it as the final step in the python script - I think that would remove that last possibility of a partial write.

What do you think?

Reply:
Yes, I think that is a safer approach. Even though this failure should be rare as you mentioned it, I think it is better to be safe than sorry here

Reply:
That makes sense! I'll do this then

Reply:
Updated!